[Nasya Myrilla Wijaya] iP

myrilla12 opened the pull request:
# Eevee Chatbot
*Named after my dog* 🦮 
> This is my original quote

- easy to use
- efficient user interface

All you need to do is:
1. **Download** it
2. Open it
3. Type in commands
4. Make a task list ~~yay~~

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
This is not actually my `main` class. 

Things to do: 
- [ ] Make a GUI
- [ ] Reminders

I learned how to do this from [Github documentation](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

myrilla12 committed:
Rename Duke

Duke is a placeholder name.

Should be changed to become more distinct.

Let's rename Duke, change greeting string and add exit string.

myrilla12 committed:
Make chatbot echo

When user inputs a string, the chatbot should print the same string.

myrilla12 committed:
Update Eevee to exit on bye

myrilla12 committed:
Update Eevee to exit on bye

myrilla12 committed:
Update Eevee to list tasks

myrilla12 committed:
Update format and print statements

The format is not organised due to blank lines. Print statements are a bit ambiguous and could use more detail.

Update format to remove blank spaces. Rewrite 'added: ' print statement to include more detail.

myrilla12 committed:
Create object Task

Chatbot saves task as string.

Not sustainable as we need task to have other attributes now such as being done or not done.

Update chatbot such that it creates Task object for each task instead.

myrilla12 committed:
Add mark and unmark task functionality

myrilla12 committed:
Create subclasses for Task object

Only supports one kind of task.

Add Deadline, Event and Todo children classes of Task.

myrilla12 committed:
Modify chatbot to respond to each type of task

myrilla12 committed:
Create sample inputs and expected outputs for testing

myrilla12 committed:
Handle exception if no task description given

myrilla12 committed:
Handle the case if there are no tasks

myrilla12 committed:
Handle exceptions for user input and commands

Wrong inputs for different task types cause errors but don't result in explanatory error messages.

Handle the exceptions with customised error messages to make it more user friendly.

Handle when user tries to mark a marked task or unmark an unmarked test. Informs user it has already been marked/unmarked instead of returning same string.

myrilla12 committed:
Update Eevee to be able to delete tasks

Handle exception if task number does not exist.

myrilla12 committed:
Enumerate chatbot

Code is convoluted and does not have good readability.

Enumerate commands to handle them in a more organised manner.

myrilla12 committed:
Add JavaDoc comments

myrilla12 committed:
Store tasks when chatbot exits

myrilla12 committed:
Split lines over 120 chars

myrilla12 committed:
Save and load tasks in a file

myrilla12 committed:
Merge branch 'branch-Level-7'

myrilla12 committed:
Format dates for deadline tasks

myrilla12 committed:
Move date format to deadline class

myrilla12 committed:
Stop tracking tasks.txt

myrilla12 committed:
Revert "Stop tracking tasks.txt"

This reverts commit 5c86b5b0ec25cbde9a08f357928c3c0c8f02f295.

myrilla12 committed:
Format dates for event tasks

myrilla12 committed:
Handle exception if from date comes after to date

myrilla12 committed:
Update exception message

myrilla12 committed:
Refactor code to become more OOP

myrilla12 committed:
Tidy task string descriptions

myrilla12 committed:
Update so storage writes over the file at every change

myrilla12 committed:
Organize classes into java packages

myrilla12 committed:
Merge branch 'add-gradle-support'

myrilla12 committed:
Update gradle build to run eevee

myrilla12 committed:
Create test for Task class

myrilla12 committed:
Correct import statements

myrilla12 committed:
Fix error handling for empty inputs

myrilla12 committed:
Create tests for TaskList class

myrilla12 committed:
Update gradle to create jar file

myrilla12 committed:
Update storage to auto create storage file

myrilla12 committed:
Add JavaDoc comments

myrilla12 committed:
Add function to find tasks in an arraylist of tasks

myrilla12 committed:
Update eevee to handle command 'find'

myrilla12 committed:
Update task counter string

myrilla12 committed:
Add in line comments

myrilla12 committed:
Merge branch 'branch-Level-9'

myrilla12 committed:
Merge branch 'branch-A-JavaDoc'

InfinityTwo commented on src/main/java/eevee/Eevee.java:
```diff
@@ -0,0 +1,161 @@
+            try {
+                Parser.Command command = parser.parseCommand(input);
+
+                switch (command) {
```
Using t, s, d, e as names in the entire switch statement can be a little troublesome to remember since there is an increasing number of single lettered variables. Would writing it in full be better?

InfinityTwo commented on src/main/java/eevee/Event.java:
```diff
@@ -0,0 +1,56 @@
+        String endDate = to.trim();
+        try {
+            LocalDate date = LocalDate.parse(startDate);
+            LocalDate date2 = LocalDate.parse(endDate);
```
Would it better to write endDateParsed instead of date2? same with date as well

InfinityTwo commented on src/main/java/eevee/Ui.java:
```diff
@@ -0,0 +1,59 @@
+ * Represents the CLI UI for the application.
+ */
+public class Ui {
+    private final String divider = "____________________________________________________________\n";
```
Shouldn't final var names be in SCREAMING_SNAKE_CASE?

InfinityTwo reviewed (commented):
I see a lot of single character variable names, which may be a bit annoying to follow but those variables are rather straightforward so it's hard to say if it's better to change it. Other than that, the rest LGTM.

warg-void reviewed (commented):
LGTM!

warg-void commented on src/main/java/eevee/Eevee.java:
I like the use of try catch blocks. In the main run loop could the try catch block be split to cover just the input command?

warg-void commented on src/main/java/eevee/Deadline.java:
Could by in the Deadline constructor be given a different name as it is unformatted while the by in the class is formatted?

myrilla12 committed:
Fix typo

myrilla12 committed:
Update build gradle

myrilla12 committed:
Make basic GUI work

myrilla12 committed:
Modify Eevee to work with GUI

myrilla12 committed:
Modify styling of chatbot

myrilla12 committed:
Add assertions to class Eevee

No assertions used, no verification to avoid bugs.

Adding assertions improves chances of catching and solving bugs.

Let's add assertions at various points in the code to document important assumptions that should hold true.

myrilla12 committed:
Add assertions

myrilla12 committed:
Refactor getResponse method

Method is very long and does not fully follow SLAP due to all the command handling being thrown into the switch case block.

Extracting methods improves the code quality.

As a step towards improving the code quality, let's extract all the command handling into separate methods for each command. Doing so will make the method shorter and improve readability as well as ensure adherence to SLAP.

myrilla12 committed:
Refactor Deadline class

Deadline class constructor has 2 levels of abstracted.

Refactoring the constructor will ensure adherence to SLAP to improve code quality.

Let's extract method for formatting due date out of the constructor.

myrilla12 committed:
Add JavaDoc comments

myrilla12 committed:
Refactor Event class

myrilla12 committed:
Refactor Storage class

Storage class has a load task method that is long and does not adhere to SLAP.

Extracting methods for handling loading of different types of tasks will improve code quality.

Let's extract methods to handle todo, deadline and events tasks out of the loadTask method.

myrilla12 committed:
Merge pull request #1 from myrilla12/branch-A-Assertions

Add assertions

myrilla12 committed:
Merge branch 'master' of https://github.com/myrilla12/ip

myrilla12 committed:
Merge branch 'master' into branch-A-CodeQuality

myrilla12 committed:
Merge pull request #2 from myrilla12/branch-A-CodeQuality

Improve code quality

myrilla12 committed:
Create main.yml

myrilla12 committed:
Update Task to have priority attribute

myrilla12 committed:
Store task priority

myrilla12 committed:
Allow users to set priority of tasks

myrilla12 committed:
Allow users to list tasks by priority level

myrilla12 committed:
Handle when no task of certain priority found

myrilla12 committed:
Tidy code

myrilla12 committed:
Add greeting string

myrilla12 committed:
Add UI image

myrilla12 committed:
Update readme

myrilla12 committed:
Update README

myrilla12 committed:
Display error message in GUI

myrilla12 committed:
Update README

myrilla12 committed:
Update README

myrilla12 committed:
Handle invalid priority level input

myrilla12 committed:
Add command summary to README

myrilla12 committed:
Refactor project structure

myrilla12 committed:
Add EventTest

myrilla12 committed:
Update MainWindow

myrilla12 committed:
Fix jar bug
